x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_icon_property.tsx

```diff
@@ -33,12 +37,23 @@ export class DynamicIconProperty extends DynamicStyleProperty<IconDynamicOptions
 
   syncIconWithMb(symbolLayerId: string, mbMap: MbMap, iconPixelSize: number) {
     if (this._isIconDynamicConfigComplete()) {
-      mbMap.setLayoutProperty(
-        symbolLayerId,
-        'icon-image',
-        this._getMbIconImageExpression(iconPixelSize)
-      );
-      mbMap.setLayoutProperty(symbolLayerId, 'icon-anchor', this._getMbIconAnchorExpression());
+      const { customIconStops, iconPaletteId } = this._options;
+      const icons = [
+        ...(iconPaletteId ? getIconPalette(iconPaletteId) : []),
+        ...(customIconStops ? customIconStops.map(({ icon }) => icon) : []),
+      ];
+      Promise.all(
```

Comment:
Discussed offline with @nickpeihl @nreese . 

Would move the registration of the icons to the bootstrap phase when the maps get initialized. Basically, similar to how it used to be.

+1 on keeping `syncIconWithMb` synchronous and not introduce any callbacks or other async-code in here.